repo/fsrepo/datastores.go

```diff
@@ -136,6 +147,10 @@ func FlatfsDatastoreConfig(params map[string]interface{}) (DatastoreConfig, erro
 	return &c, nil
 }
 
+func (c *flatfsDatastoreConfig) DiskId() string {
+	return fmt.Sprintf("flatfs;%s;%s", c.path, c.shardFun.String())
```

Comment:
The path helps to identify the datastore.  Especially if we ever allow multiple datastores on the same mountpoint (like a unionfs).  Also see my comment I just left.